// @schedule-x/vue/dist/schedule-x-vue.mjs
import { h as Q, defineComponent as _t, isReactive as Gt, Teleport as Zt, Fragment as qt, watch as Xt, onMounted as Qt } from "vue";
const en = (e, t) => (n, r) => {
  const i = {
    Component: Q(t, r),
    wrapperElement: n
  };
  e(i);
};
class tn extends Error {
  constructor(t) {
    super(t), this.name = "[Schedule-X reactivity error]";
  }
}
const go = _t({
  name: "ScheduleXCalendar",
  props: {
    calendarApp: {
      type: Object,
      required: !0
    },
    customComponents: {
      type: Object,
      default: () => ({})
    }
  },
  data() {
    return {
      elId: "sx" + Math.random().toString(36).substr(2, 9),
      customComponentsMeta: []
    };
  },
  mounted() {
    if (Gt(this.calendarApp))
      throw new tn("calendarApp cannot be saved in a ref. Since this causes deep reactivity, it destroys the calendars internal reactivity. Save in a normal const or shallowRef");
    const e = {
      ...this.customComponents,
      ...this.$slots
    };
    for (const [t, n] of Object.entries(e))
      n && this.calendarApp._setCustomComponentFn(t, en(this.setCustomComponentMeta, n));
    this.calendarApp.render(document.getElementById(this.elId));
  },
  unmounted() {
    this.calendarApp.destroy();
  },
  methods: {
    setCustomComponentMeta(e) {
      if (!(e.wrapperElement instanceof HTMLElement))
        return;
      const n = ({ wrapperElement: a }) => a instanceof HTMLElement, r = [
        ...this.customComponentsMeta.filter(n)
      ], i = e.wrapperElement.dataset.ccid, o = r.find(({ wrapperElement: a }) => a.dataset.ccid === i);
      o && r.splice(r.indexOf(o), 1), this.customComponentsMeta = [...r, e];
    }
  },
  render() {
    const e = this.customComponentsMeta.map(({ Component: t, wrapperElement: n }) => Q(Zt, { to: n }, t));
    return Q("div", {
      id: this.elId,
      class: "sx-vue-calendar-wrapper"
    }, Q(qt, {}, e));
  }
});
var ve, h, ft, ht, V, Ve, pt, Se, vt, Ae, Me, Ye, mt, te = {}, Dt = [], nn = /acit|ex(?:s|g|n|p|$)|rph|grid|ows|mnc|ntw|ine[ch]|zoo|^ord|itera/i, me = Array.isArray;
function F(e, t) {
  for (var n in t) e[n] = t[n];
  return e;
}
function We(e) {
  e && e.parentNode && e.parentNode.removeChild(e);
}
function z(e, t, n) {
  var r, i, o, a = {};
  for (o in t) o == "key" ? r = t[o] : o == "ref" ? i = t[o] : a[o] = t[o];
  if (arguments.length > 2 && (a.children = arguments.length > 3 ? ve.call(arguments, 2) : n), typeof e == "function" && e.defaultProps != null) for (o in e.defaultProps) a[o] === void 0 && (a[o] = e.defaultProps[o]);
  return de(e, a, r, i, null);
}
function de(e, t, n, r, i) {
  var o = { type: e, props: t, key: n, ref: r, __k: null, __: null, __b: 0, __e: null, __c: null, constructor: void 0, __v: i ?? ++ft, __i: -1, __u: 0 };
  return i == null && h.vnode != null && h.vnode(o), o;
}
function Y(e) {
  return e.children;
}
function O(e, t) {
  this.props = e, this.context = t;
}
function K(e, t) {
  if (t == null) return e.__ ? K(e.__, e.__i + 1) : null;
  for (var n; t < e.__k.length; t++) if ((n = e.__k[t]) != null && n.__e != null) return n.__e;
  return typeof e.type == "function" ? K(e) : null;
}
function kt(e) {
  var t, n;
  if ((e = e.__) != null && e.__c != null) {
    for (e.__e = e.__c.base = null, t = 0; t < e.__k.length; t++) if ((n = e.__k[t]) != null && n.__e != null) {
      e.__e = e.__c.base = n.__e;
      break;
    }
    return kt(e);
  }
}
function Ne(e) {
  (!e.__d && (e.__d = !0) && V.push(e) && !_e.__r++ || Ve !== h.debounceRendering) && ((Ve = h.debounceRendering) || pt)(_e);
}
function _e() {
  var e, t, n, r, i, o, a, s;
  for (V.sort(Se); e = V.shift(); ) e.__d && (t = V.length, r = void 0, o = (i = (n = e).__v).__e, a = [], s = [], n.__P && ((r = F({}, i)).__v = i.__v + 1, h.vnode && h.vnode(r), Fe(n.__P, r, i, n.__n, n.__P.namespaceURI, 32 & i.__u ? [o] : null, a, o ?? K(i), !!(32 & i.__u), s), r.__v = i.__v, r.__.__k[r.__i] = r, bt(a, r, s), r.__e != o && kt(r)), V.length > t && V.sort(Se));
  _e.__r = 0;
}
function yt(e, t, n, r, i, o, a, s, d, l, _) {
  var u, c, f, m, v, P, k = r && r.__k || Dt, g = t.length;
  for (d = rn(n, t, k, d, g), u = 0; u < g; u++) (f = n.__k[u]) != null && (c = f.__i === -1 ? te : k[f.__i] || te, f.__i = u, P = Fe(e, f, c, i, o, a, s, d, l, _), m = f.__e, f.ref && c.ref != f.ref && (c.ref && je(c.ref, null, f), _.push(f.ref, f.__c || m, f)), v == null && m != null && (v = m), 4 & f.__u || c.__k === f.__k ? d = gt(f, d, e) : typeof f.type == "function" && P !== void 0 ? d = P : m && (d = m.nextSibling), f.__u &= -7);
  return n.__e = v, d;
}
function rn(e, t, n, r, i) {
  var o, a, s, d, l, _ = n.length, u = _, c = 0;
  for (e.__k = new Array(i), o = 0; o < i; o++) (a = t[o]) != null && typeof a != "boolean" && typeof a != "function" ? (d = o + c, (a = e.__k[o] = typeof a == "string" || typeof a == "number" || typeof a == "bigint" || a.constructor == String ? de(null, a, null, null, null) : me(a) ? de(Y, { children: a }, null, null, null) : a.constructor === void 0 && a.__b > 0 ? de(a.type, a.props, a.key, a.ref ? a.ref : null, a.__v) : a).__ = e, a.__b = e.__b + 1, s = null, (l = a.__i = on(a, n, d, u)) !== -1 && (u--, (s = n[l]) && (s.__u |= 2)), s == null || s.__v === null ? (l == -1 && c--, typeof a.type != "function" && (a.__u |= 4)) : l != d && (l == d - 1 ? c-- : l == d + 1 ? c++ : (l > d ? c-- : c++, a.__u |= 4))) : e.__k[o] = null;
  if (u) for (o = 0; o < _; o++) (s = n[o]) != null && !(2 & s.__u) && (s.__e == r && (r = K(s)), Pt(s, s));
  return r;
}
function gt(e, t, n) {
  var r, i;
  if (typeof e.type == "function") {
    for (r = e.__k, i = 0; r && i < r.length; i++) r[i] && (r[i].__ = e, t = gt(r[i], t, n));
    return t;
  }
  e.__e != t && (t && e.type && !n.contains(t) && (t = K(e)), n.insertBefore(e.__e, t || null), t = e.__e);
  do
    t = t && t.nextSibling;
  while (t != null && t.nodeType == 8);
  return t;
}
function fe(e, t) {
  return t = t || [], e == null || typeof e == "boolean" || (me(e) ? e.some(function(n) {
    fe(n, t);
  }) : t.push(e)), t;
}
function on(e, t, n, r) {
  var i, o, a = e.key, s = e.type, d = t[n];
  if (d === null || d && a == d.key && s === d.type && !(2 & d.__u)) return n;
  if (r > (d != null && !(2 & d.__u) ? 1 : 0)) for (i = n - 1, o = n + 1; i >= 0 || o < t.length; ) {
    if (i >= 0) {
      if ((d = t[i]) && !(2 & d.__u) && a == d.key && s === d.type) return i;
      i--;
    }
    if (o < t.length) {
      if ((d = t[o]) && !(2 & d.__u) && a == d.key && s === d.type) return o;
      o++;
    }
  }
  return -1;
}
function Ue(e, t, n) {
  t[0] == "-" ? e.setProperty(t, n ?? "") : e[t] = n == null ? "" : typeof n != "number" || nn.test(t) ? n : n + "px";
}
function ae(e, t, n, r, i) {
  var o;
  e: if (t == "style") if (typeof n == "string") e.style.cssText = n;
  else {
    if (typeof r == "string" && (e.style.cssText = r = ""), r) for (t in r) n && t in n || Ue(e.style, t, "");
    if (n) for (t in n) r && n[t] === r[t] || Ue(e.style, t, n[t]);
  }
  else if (t[0] == "o" && t[1] == "n") o = t != (t = t.replace(vt, "$1")), t = t.toLowerCase() in e || t == "onFocusOut" || t == "onFocusIn" ? t.toLowerCase().slice(2) : t.slice(2), e.l || (e.l = {}), e.l[t + o] = n, n ? r ? n.u = r.u : (n.u = Ae, e.addEventListener(t, o ? Ye : Me, o)) : e.removeEventListener(t, o ? Ye : Me, o);
  else {
    if (i == "http://www.w3.org/2000/svg") t = t.replace(/xlink(H|:h)/, "h").replace(/sName$/, "s");
    else if (t != "width" && t != "height" && t != "href" && t != "list" && t != "form" && t != "tabIndex" && t != "download" && t != "rowSpan" && t != "colSpan" && t != "role" && t != "popover" && t in e) try {
      e[t] = n ?? "";
      break e;
    } catch {
    }
    typeof n == "function" || (n == null || n === !1 && t[4] != "-" ? e.removeAttribute(t) : e.setAttribute(t, t == "popover" && n == 1 ? "" : n));
  }
}
function He(e) {
  return function(t) {
    if (this.l) {
      var n = this.l[t.type + e];
      if (t.t == null) t.t = Ae++;
      else if (t.t < n.u) return;
      return n(h.event ? h.event(t) : t);
    }
  };
}
function Fe(e, t, n, r, i, o, a, s, d, l) {
  var _, u, c, f, m, v, P, k, g, Z, W, ie, q, Re, oe, ye, ge, T = t.type;
  if (t.constructor !== void 0) return null;
  128 & n.__u && (d = !!(32 & n.__u), o = [s = t.__e = n.__e]), (_ = h.__b) && _(t);
  e: if (typeof T == "function") try {
    if (k = t.props, g = "prototype" in T && T.prototype.render, Z = (_ = T.contextType) && r[_.__c], W = _ ? Z ? Z.props.value : _.__ : r, n.__c ? P = (u = t.__c = n.__c).__ = u.__E : (g ? t.__c = u = new T(k, W) : (t.__c = u = new O(k, W), u.constructor = T, u.render = sn), Z && Z.sub(u), u.props = k, u.state || (u.state = {}), u.context = W, u.__n = r, c = u.__d = !0, u.__h = [], u._sb = []), g && u.__s == null && (u.__s = u.state), g && T.getDerivedStateFromProps != null && (u.__s == u.state && (u.__s = F({}, u.__s)), F(u.__s, T.getDerivedStateFromProps(k, u.__s))), f = u.props, m = u.state, u.__v = t, c) g && T.getDerivedStateFromProps == null && u.componentWillMount != null && u.componentWillMount(), g && u.componentDidMount != null && u.__h.push(u.componentDidMount);
    else {
      if (g && T.getDerivedStateFromProps == null && k !== f && u.componentWillReceiveProps != null && u.componentWillReceiveProps(k, W), !u.__e && (u.shouldComponentUpdate != null && u.shouldComponentUpdate(k, u.__s, W) === !1 || t.__v == n.__v)) {
        for (t.__v != n.__v && (u.props = k, u.state = u.__s, u.__d = !1), t.__e = n.__e, t.__k = n.__k, t.__k.some(function(X) {
          X && (X.__ = t);
        }), ie = 0; ie < u._sb.length; ie++) u.__h.push(u._sb[ie]);
        u._sb = [], u.__h.length && a.push(u);
        break e;
      }
      u.componentWillUpdate != null && u.componentWillUpdate(k, u.__s, W), g && u.componentDidUpdate != null && u.__h.push(function() {
        u.componentDidUpdate(f, m, v);
      });
    }
    if (u.context = W, u.props = k, u.__P = e, u.__e = !1, q = h.__r, Re = 0, g) {
      for (u.state = u.__s, u.__d = !1, q && q(t), _ = u.render(u.props, u.state, u.context), oe = 0; oe < u._sb.length; oe++) u.__h.push(u._sb[oe]);
      u._sb = [];
    } else do
      u.__d = !1, q && q(t), _ = u.render(u.props, u.state, u.context), u.state = u.__s;
    while (u.__d && ++Re < 25);
    u.state = u.__s, u.getChildContext != null && (r = F(F({}, r), u.getChildContext())), g && !c && u.getSnapshotBeforeUpdate != null && (v = u.getSnapshotBeforeUpdate(f, m)), s = yt(e, me(ye = _ != null && _.type === Y && _.key == null ? _.props.children : _) ? ye : [ye], t, n, r, i, o, a, s, d, l), u.base = t.__e, t.__u &= -161, u.__h.length && a.push(u), P && (u.__E = u.__ = null);
  } catch (X) {
    if (t.__v = null, d || o != null) if (X.then) {
      for (t.__u |= d ? 160 : 128; s && s.nodeType == 8 && s.nextSibling; ) s = s.nextSibling;
      o[o.indexOf(s)] = null, t.__e = s;
    } else for (ge = o.length; ge--; ) We(o[ge]);
    else t.__e = n.__e, t.__k = n.__k;
    h.__e(X, t, n);
  }
  else o == null && t.__v == n.__v ? (t.__k = n.__k, t.__e = n.__e) : s = t.__e = an(n.__e, t, n, r, i, o, a, d, l);
  return (_ = h.diffed) && _(t), 128 & t.__u ? void 0 : s;
}
function bt(e, t, n) {
  for (var r = 0; r < n.length; r++) je(n[r], n[++r], n[++r]);
  h.__c && h.__c(t, e), e.some(function(i) {
    try {
      e = i.__h, i.__h = [], e.some(function(o) {
        o.call(i);
      });
    } catch (o) {
      h.__e(o, i.__v);
    }
  });
}
function an(e, t, n, r, i, o, a, s, d) {
  var l, _, u, c, f, m, v, P = n.props, k = t.props, g = t.type;
  if (g == "svg" ? i = "http://www.w3.org/2000/svg" : g == "math" ? i = "http://www.w3.org/1998/Math/MathML" : i || (i = "http://www.w3.org/1999/xhtml"), o != null) {
    for (l = 0; l < o.length; l++) if ((f = o[l]) && "setAttribute" in f == !!g && (g ? f.localName == g : f.nodeType == 3)) {
      e = f, o[l] = null;
      break;
    }
  }
  if (e == null) {
    if (g == null) return document.createTextNode(k);
    e = document.createElementNS(i, g, k.is && k), s && (h.__m && h.__m(t, o), s = !1), o = null;
  }
  if (g === null) P === k || s && e.data === k || (e.data = k);
  else {
    if (o = o && ve.call(e.childNodes), P = n.props || te, !s && o != null) for (P = {}, l = 0; l < e.attributes.length; l++) P[(f = e.attributes[l]).name] = f.value;
    for (l in P) if (f = P[l], l != "children") {
      if (l == "dangerouslySetInnerHTML") u = f;
      else if (!(l in k)) {
        if (l == "value" && "defaultValue" in k || l == "checked" && "defaultChecked" in k) continue;
        ae(e, l, null, f, i);
      }
    }
    for (l in k) f = k[l], l == "children" ? c = f : l == "dangerouslySetInnerHTML" ? _ = f : l == "value" ? m = f : l == "checked" ? v = f : s && typeof f != "function" || P[l] === f || ae(e, l, f, P[l], i);
    if (_) s || u && (_.__html === u.__html || _.__html === e.innerHTML) || (e.innerHTML = _.__html), t.__k = [];
    else if (u && (e.innerHTML = ""), yt(e, me(c) ? c : [c], t, n, r, g == "foreignObject" ? "http://www.w3.org/1999/xhtml" : i, o, a, o ? o[0] : n.__k && K(n, 0), s, d), o != null) for (l = o.length; l--; ) We(o[l]);
    s || (l = "value", g == "progress" && m == null ? e.removeAttribute("value") : m !== void 0 && (m !== e[l] || g == "progress" && !m || g == "option" && m !== P[l]) && ae(e, l, m, P[l], i), l = "checked", v !== void 0 && v !== e[l] && ae(e, l, v, P[l], i));
  }
  return e;
}
function je(e, t, n) {
  try {
    if (typeof e == "function") {
      var r = typeof e.__u == "function";
      r && e.__u(), r && t == null || (e.__u = e(t));
    } else e.current = t;
  } catch (i) {
    h.__e(i, n);
  }
}
function Pt(e, t, n) {
  var r, i;
  if (h.unmount && h.unmount(e), (r = e.ref) && (r.current && r.current !== e.__e || je(r, null, t)), (r = e.__c) != null) {
    if (r.componentWillUnmount) try {
      r.componentWillUnmount();
    } catch (o) {
      h.__e(o, t);
    }
    r.base = r.__P = null;
  }
  if (r = e.__k) for (i = 0; i < r.length; i++) r[i] && Pt(r[i], t, n || typeof e.type != "function");
  n || We(e.__e), e.__c = e.__ = e.__e = void 0;
}
function sn(e, t, n) {
  return this.constructor(e, n);
}
function xe(e, t, n) {
  var r, i, o, a;
  t == document && (t = document.documentElement), h.__ && h.__(e, t), i = (r = !1) ? null : t.__k, o = [], a = [], Fe(t, e = t.__k = z(Y, null, [e]), i || te, te, t.namespaceURI, i ? null : t.firstChild ? ve.call(t.childNodes) : null, o, i ? i.__e : t.firstChild, r, a), bt(o, e, a);
}
function un(e, t) {
  var n = { __c: t = "__cC" + mt++, __: e, Consumer: function(r, i) {
    return r.children(i);
  }, Provider: function(r) {
    var i, o;
    return this.getChildContext || (i = /* @__PURE__ */ new Set(), (o = {})[t] = this, this.getChildContext = function() {
      return o;
    }, this.componentWillUnmount = function() {
      i = null;
    }, this.shouldComponentUpdate = function(a) {
      this.props.value !== a.value && i.forEach(function(s) {
        s.__e = !0, Ne(s);
      });
    }, this.sub = function(a) {
      i.add(a);
      var s = a.componentWillUnmount;
      a.componentWillUnmount = function() {
        i && i.delete(a), s && s.call(a);
      };
    }), r.children;
  } };
  return n.Provider.__ = n.Consumer.contextType = n;
}
ve = Dt.slice, h = { __e: function(e, t, n, r) {
  for (var i, o, a; t = t.__; ) if ((i = t.__c) && !i.__) try {
    if ((o = i.constructor) && o.getDerivedStateFromError != null && (i.setState(o.getDerivedStateFromError(e)), a = i.__d), i.componentDidCatch != null && (i.componentDidCatch(e, r || {}), a = i.__d), a) return i.__E = i;
  } catch (s) {
    e = s;
  }
  throw e;
} }, ft = 0, ht = function(e) {
  return e != null && e.constructor == null;
}, O.prototype.setState = function(e, t) {
  var n;
  n = this.__s != null && this.__s !== this.state ? this.__s : this.__s = F({}, this.state), typeof e == "function" && (e = e(F({}, n), this.props)), e && F(n, e), e != null && this.__v && (t && this._sb.push(t), Ne(this));
}, O.prototype.forceUpdate = function(e) {
  this.__v && (this.__e = !0, e && this.__h.push(e), Ne(this));
}, O.prototype.render = Y, V = [], pt = typeof Promise == "function" ? Promise.prototype.then.bind(Promise.resolve()) : setTimeout, Se = function(e, t) {
  return e.__v.__b - t.__v.__b;
}, _e.__r = 0, vt = /(PointerCapture)$|Capture$/i, Ae = 0, Me = He(!1), Ye = He(!0), mt = 0;
var dn = 0;
function p(e, t, n, r, i, o) {
  t || (t = {});
  var a, s, d = t;
  if ("ref" in d) for (s in d = {}, t) s == "ref" ? a = t[s] : d[s] = t[s];
  var l = { type: e, props: d, key: n, ref: a, __k: null, __: null, __b: 0, __e: null, __c: null, constructor: void 0, __v: --dn, __i: -1, __u: 0, __source: i, __self: o };
  if (typeof e == "function" && (a = e.defaultProps)) for (s in a) d[s] === void 0 && (d[s] = a[s]);
  return h.vnode && h.vnode(l), l;
}
var B, b, be, ze, Le = 0, wt = [], w = h, Ke = w.__b, Be = w.__r, Je = w.diffed, Ge = w.__c, Ze = w.unmount, qe = w.__;
function De(e, t) {
  w.__h && w.__h(b, e, Le || t), Le = 0;
  var n = b.__H || (b.__H = { __: [], __h: [] });
  return e >= n.__.length && n.__.push({}), n.__[e];
}
function $(e) {
  return Le = 1, ln(Yt, e);
}
function ln(e, t, n) {
  var r = De(B++, 2);
  if (r.t = e, !r.__c && (r.__ = [n ? n(t) : Yt(void 0, t), function(s) {
    var d = r.__N ? r.__N[0] : r.__[0], l = r.t(d, s);
    d !== l && (r.__N = [l, r.__[1]], r.__c.setState({}));
  }], r.__c = b, !b.u)) {
    var i = function(s, d, l) {
      if (!r.__c.__H) return !0;
      var _ = r.__c.__H.__.filter(function(c) {
        return !!c.__c;
      });
      if (_.every(function(c) {
        return !c.__N;
      })) return !o || o.call(this, s, d, l);
      var u = r.__c.props !== s;
      return _.forEach(function(c) {
        if (c.__N) {
          var f = c.__[0];
          c.__ = c.__N, c.__N = void 0, f !== c.__[0] && (u = !0);
        }
      }), o && o.call(this, s, d, l) || u;
    };
    b.u = !0;
    var o = b.shouldComponentUpdate, a = b.componentWillUpdate;
    b.componentWillUpdate = function(s, d, l) {
      if (this.__e) {
        var _ = o;
        o = void 0, i(s, d, l), o = _;
      }
      a && a.call(this, s, d, l);
    }, b.shouldComponentUpdate = i;
  }
  return r.__N || r.__;
}
function C(e, t) {
  var n = De(B++, 3);
  !w.__s && Mt(n.__H, t) && (n.__ = e, n.i = t, b.__H.__h.push(n));
}
function St(e, t) {
  var n = De(B++, 7);
  return Mt(n.__H, t) && (n.__ = e(), n.__H = t, n.__h = e), n.__;
}
function R(e) {
  var t = b.context[e.__c], n = De(B++, 9);
  return n.c = e, t ? (n.__ == null && (n.__ = !0, t.sub(b)), t.props.value) : e.__;
}
function cn() {
  for (var e; e = wt.shift(); ) if (e.__P && e.__H) try {
    e.__H.__h.forEach(le), e.__H.__h.forEach(Ee), e.__H.__h = [];
  } catch (t) {
    e.__H.__h = [], w.__e(t, e.__v);
  }
}
w.__b = function(e) {
  b = null, Ke && Ke(e);
}, w.__ = function(e, t) {
  e && t.__k && t.__k.__m && (e.__m = t.__k.__m), qe && qe(e, t);
}, w.__r = function(e) {
  Be && Be(e), B = 0;
  var t = (b = e.__c).__H;
  t && (be === b ? (t.__h = [], b.__h = [], t.__.forEach(function(n) {
    n.__N && (n.__ = n.__N), n.i = n.__N = void 0;
  })) : (t.__h.forEach(le), t.__h.forEach(Ee), t.__h = [], B = 0)), be = b;
}, w.diffed = function(e) {
  Je && Je(e);
  var t = e.__c;
  t && t.__H && (t.__H.__h.length && (wt.push(t) !== 1 && ze === w.requestAnimationFrame || ((ze = w.requestAnimationFrame) || _n)(cn)), t.__H.__.forEach(function(n) {
    n.i && (n.__H = n.i), n.i = void 0;
  })), be = b = null;
}, w.__c = function(e, t) {
  t.some(function(n) {
    try {
      n.__h.forEach(le), n.__h = n.__h.filter(function(r) {
        return !r.__ || Ee(r);
      });
    } catch (r) {
      t.some(function(i) {
        i.__h && (i.__h = []);
      }), t = [], w.__e(r, n.__v);
    }
  }), Ge && Ge(e, t);
}, w.unmount = function(e) {
  Ze && Ze(e);
  var t, n = e.__c;
  n && n.__H && (n.__H.__.forEach(function(r) {
    try {
      le(r);
    } catch (i) {
      t = i;
    }
  }), n.__H = void 0, t && w.__e(t, n.__v));
};
var Xe = typeof requestAnimationFrame == "function";
function _n(e) {
  var t, n = function() {
    clearTimeout(r), Xe && cancelAnimationFrame(t), setTimeout(e);
  }, r = setTimeout(n, 100);
  Xe && (t = requestAnimationFrame(n));
}
function le(e) {
  var t = b, n = e.__c;
  typeof n == "function" && (e.__c = void 0, n()), b = t;
}
function Ee(e) {
  var t = b;
  e.__c = e.__(), b = t;
}
function Mt(e, t) {
  return !e || e.length !== t.length || t.some(function(n, r) {
    return n !== e[r];
  });
}
function Yt(e, t) {
  return typeof t == "function" ? t(e) : t;
}
function fn(e, t) {
  for (var n in t) e[n] = t[n];
  return e;
}
function Qe(e, t) {
  for (var n in e) if (n !== "__source" && !(n in t)) return !0;
  for (var r in t) if (r !== "__source" && e[r] !== t[r]) return !0;
  return !1;
}
function et(e, t) {
  this.props = e, this.context = t;
}
(et.prototype = new O()).isPureReactComponent = !0, et.prototype.shouldComponentUpdate = function(e, t) {
  return Qe(this.props, e) || Qe(this.state, t);
};
var tt = h.__b;
h.__b = function(e) {
  e.type && e.type.__f && e.ref && (e.props.ref = e.ref, e.ref = null), tt && tt(e);
};
var hn = h.__e;
h.__e = function(e, t, n, r) {
  if (e.then) {
    for (var i, o = t; o = o.__; ) if ((i = o.__c) && i.__c) return t.__e == null && (t.__e = n.__e, t.__k = n.__k), i.__c(e, t);
  }
  hn(e, t, n, r);
};
var nt = h.unmount;
function Nt(e, t, n) {
  return e && (e.__c && e.__c.__H && (e.__c.__H.__.forEach(function(r) {
    typeof r.__c == "function" && r.__c();
  }), e.__c.__H = null), (e = fn({}, e)).__c != null && (e.__c.__P === n && (e.__c.__P = t), e.__c = null), e.__k = e.__k && e.__k.map(function(r) {
    return Nt(r, t, n);
  })), e;
}
function xt(e, t, n) {
  return e && n && (e.__v = null, e.__k = e.__k && e.__k.map(function(r) {
    return xt(r, t, n);
  }), e.__c && e.__c.__P === t && (e.__e && n.appendChild(e.__e), e.__c.__e = !0, e.__c.__P = n)), e;
}
function Pe() {
  this.__u = 0, this.o = null, this.__b = null;
}
function Lt(e) {
  var t = e.__.__c;
  return t && t.__a && t.__a(e);
}
function se() {
  this.i = null, this.l = null;
}
h.unmount = function(e) {
  var t = e.__c;
  t && t.__R && t.__R(), t && 32 & e.__u && (e.type = null), nt && nt(e);
}, (Pe.prototype = new O()).__c = function(e, t) {
  var n = t.__c, r = this;
  r.o == null && (r.o = []), r.o.push(n);
  var i = Lt(r.__v), o = !1, a = function() {
    o || (o = !0, n.__R = null, i ? i(s) : s());
  };
  n.__R = a;
  var s = function() {
    if (!--r.__u) {
      if (r.state.__a) {
        var d = r.state.__a;
        r.__v.__k[0] = xt(d, d.__c.__P, d.__c.__O);
      }
      var l;
      for (r.setState({ __a: r.__b = null }); l = r.o.pop(); ) l.forceUpdate();
    }
  };
  r.__u++ || 32 & t.__u || r.setState({ __a: r.__b = r.__v.__k[0] }), e.then(a, a);
}, Pe.prototype.componentWillUnmount = function() {
  this.o = [];
}, Pe.prototype.render = function(e, t) {
  if (this.__b) {
    if (this.__v.__k) {
      var n = document.createElement("div"), r = this.__v.__k[0].__c;
      this.__v.__k[0] = Nt(this.__b, n, r.__O = r.__P);
    }
    this.__b = null;
  }
  var i = t.__a && z(Y, null, e.fallback);
  return i && (i.__u &= -33), [z(Y, null, t.__a ? null : e.children), i];
};
var rt = function(e, t, n) {
  if (++n[1] === n[0] && e.l.delete(t), e.props.revealOrder && (e.props.revealOrder[0] !== "t" || !e.l.size)) for (n = e.i; n; ) {
    for (; n.length > 3; ) n.pop()();
    if (n[1] < n[0]) break;
    e.i = n = n[2];
  }
};
function pn(e) {
  return this.getChildContext = function() {
    return e.context;
  }, e.children;
}
function vn(e) {
  var t = this, n = e.h;
  t.componentWillUnmount = function() {
    xe(null, t.v), t.v = null, t.h = null;
  }, t.h && t.h !== n && t.componentWillUnmount(), t.v || (t.h = n, t.v = { nodeType: 1, parentNode: n, childNodes: [], contains: function() {
    return !0;
  }, appendChild: function(r) {
    this.childNodes.push(r), t.h.appendChild(r);
  }, insertBefore: function(r, i) {
    this.childNodes.push(r), t.h.insertBefore(r, i);
  }, removeChild: function(r) {
    this.childNodes.splice(this.childNodes.indexOf(r) >>> 1, 1), t.h.removeChild(r);
  } }), xe(z(pn, { context: t.context }, e.__v), t.v);
}
function mn(e, t) {
  var n = z(vn, { __v: e, h: t });
  return n.containerInfo = t, n;
}
(se.prototype = new O()).__a = function(e) {
  var t = this, n = Lt(t.__v), r = t.l.get(e);
  return r[0]++, function(i) {
    var o = function() {
      t.props.revealOrder ? (r.push(i), rt(t, e, r)) : i();
    };
    n ? n(o) : o();
  };
}, se.prototype.render = function(e) {
  this.i = null, this.l = /* @__PURE__ */ new Map();
  var t = fe(e.children);
  e.revealOrder && e.revealOrder[0] === "b" && t.reverse();
  for (var n = t.length; n--; ) this.l.set(t[n], this.i = [1, 0, this.i]);
  return e.children;
}, se.prototype.componentDidUpdate = se.prototype.componentDidMount = function() {
  var e = this;
  this.l.forEach(function(t, n) {
    rt(e, n, t);
  });
};
var Dn = typeof Symbol < "u" && Symbol.for && Symbol.for("react.element") || 60103, kn = /^(?:accent|alignment|arabic|baseline|cap|clip(?!PathU)|color|dominant|fill|flood|font|glyph(?!R)|horiz|image(!S)|letter|lighting|marker(?!H|W|U)|overline|paint|pointer|shape|stop|strikethrough|stroke|text(?!L)|transform|underline|unicode|units|v|vector|vert|word|writing|x(?!C))[A-Z]/, yn = /^on(Ani|Tra|Tou|BeforeInp|Compo)/, gn = /[A-Z0-9]/g, bn = typeof document < "u", Pn = function(e) {
  return (typeof Symbol < "u" && typeof Symbol() == "symbol" ? /fil|che|rad/ : /fil|che|ra/).test(e);
};
O.prototype.isReactComponent = {}, ["componentWillMount", "componentWillReceiveProps", "componentWillUpdate"].forEach(function(e) {
  Object.defineProperty(O.prototype, e, { configurable: !0, get: function() {
    return this["UNSAFE_" + e];
  }, set: function(t) {
    Object.defineProperty(this, e, { configurable: !0, writable: !0, value: t });
  } });
});
var it = h.event;
function wn() {
}
function Sn() {
  return this.cancelBubble;
}
function Mn() {
  return this.defaultPrevented;
}
h.event = function(e) {
  return it && (e = it(e)), e.persist = wn, e.isPropagationStopped = Sn, e.isDefaultPrevented = Mn, e.nativeEvent = e;
};
var Yn = { enumerable: !1, configurable: !0, get: function() {
  return this.class;
} }, ot = h.vnode;
h.vnode = function(e) {
  typeof e.type == "string" && function(t) {
    var n = t.props, r = t.type, i = {}, o = r.indexOf("-") === -1;
    for (var a in n) {
      var s = n[a];
      if (!(a === "value" && "defaultValue" in n && s == null || bn && a === "children" && r === "noscript" || a === "class" || a === "className")) {
        var d = a.toLowerCase();
        a === "defaultValue" && "value" in n && n.value == null ? a = "value" : a === "download" && s === !0 ? s = "" : d === "translate" && s === "no" ? s = !1 : d[0] === "o" && d[1] === "n" ? d === "ondoubleclick" ? a = "ondblclick" : d !== "onchange" || r !== "input" && r !== "textarea" || Pn(n.type) ? d === "onfocus" ? a = "onfocusin" : d === "onblur" ? a = "onfocusout" : yn.test(a) && (a = d) : d = a = "oninput" : o && kn.test(a) ? a = a.replace(gn, "-$&").toLowerCase() : s === null && (s = void 0), d === "oninput" && i[a = d] && (a = "oninputCapture"), i[a] = s;
      }
    }
    r == "select" && i.multiple && Array.isArray(i.value) && (i.value = fe(n.children).forEach(function(l) {
      l.props.selected = i.value.indexOf(l.props.value) != -1;
    })), r == "select" && i.defaultValue != null && (i.value = fe(n.children).forEach(function(l) {
      l.props.selected = i.multiple ? i.defaultValue.indexOf(l.props.value) != -1 : i.defaultValue == l.props.value;
    })), n.class && !n.className ? (i.class = n.class, Object.defineProperty(i, "className", Yn)) : (n.className && !n.class || n.class && n.className) && (i.class = i.className = n.className), t.props = i;
  }(e), e.$$typeof = Dn, ot && ot(e);
};
var at = h.__r;
h.__r = function(e) {
  at && at(e), e.__c;
};
var st = h.diffed;
h.diffed = function(e) {
  st && st(e);
  var t = e.props, n = e.__e;
  n != null && e.type === "textarea" && "value" in t && t.value !== n.value && (n.value = t.value == null ? "" : t.value);
};
var Nn = Symbol.for("preact-signals");
function ke() {
  if (j > 1)
    j--;
  else {
    for (var e, t = !1; ee !== void 0; ) {
      var n = ee;
      for (ee = void 0, Ce++; n !== void 0; ) {
        var r = n.o;
        if (n.o = void 0, n.f &= -3, !(8 & n.f) && Ct(n)) try {
          n.c();
        } catch (i) {
          t || (e = i, t = !0);
        }
        n = r;
      }
    }
    if (Ce = 0, j--, t) throw e;
  }
}
function xn(e) {
  if (j > 0) return e();
  j++;
  try {
    return e();
  } finally {
    ke();
  }
}
var y = void 0, ee = void 0, j = 0, Ce = 0, he = 0;
function Et(e) {
  if (y !== void 0) {
    var t = e.n;
    if (t === void 0 || t.t !== y)
      return t = { i: 0, S: e, p: y.s, n: void 0, t: y, e: void 0, x: void 0, r: t }, y.s !== void 0 && (y.s.n = t), y.s = t, e.n = t, 32 & y.f && e.S(t), t;
    if (t.i === -1)
      return t.i = 0, t.n !== void 0 && (t.n.p = t.p, t.p !== void 0 && (t.p.n = t.n), t.p = y.s, t.n = void 0, y.s.n = t, y.s = t), t;
  }
}
function S(e) {
  this.v = e, this.i = 0, this.n = void 0, this.t = void 0;
}
S.prototype.brand = Nn;
S.prototype.h = function() {
  return !0;
};
S.prototype.S = function(e) {
  this.t !== e && e.e === void 0 && (e.x = this.t, this.t !== void 0 && (this.t.e = e), this.t = e);
};
S.prototype.U = function(e) {
  if (this.t !== void 0) {
    var t = e.e, n = e.x;
    t !== void 0 && (t.x = n, e.e = void 0), n !== void 0 && (n.e = t, e.x = void 0), e === this.t && (this.t = n);
  }
};
S.prototype.subscribe = function(e) {
  var t = this;
  return H(function() {
    var n = t.value, r = y;
    y = void 0;
    try {
      e(n);
    } finally {
      y = r;
    }
  });
};
S.prototype.valueOf = function() {
  return this.value;
};
S.prototype.toString = function() {
  return this.value + "";
};
S.prototype.toJSON = function() {
  return this.value;
};
S.prototype.peek = function() {
  var e = y;
  y = void 0;
  try {
    return this.value;
  } finally {
    y = e;
  }
};
Object.defineProperty(S.prototype, "value", { get: function() {
  var e = Et(this);
  return e !== void 0 && (e.i = this.i), this.v;
}, set: function(e) {
  if (e !== this.v) {
    if (Ce > 100) throw new Error("Cycle detected");
    this.v = e, this.i++, he++, j++;
    try {
      for (var t = this.t; t !== void 0; t = t.x) t.t.N();
    } finally {
      ke();
    }
  }
} });
function M(e) {
  return new S(e);
}
function Ct(e) {
  for (var t = e.s; t !== void 0; t = t.n) if (t.S.i !== t.i || !t.S.h() || t.S.i !== t.i) return !0;
  return !1;
}
function Tt(e) {
  for (var t = e.s; t !== void 0; t = t.n) {
    var n = t.S.n;
    if (n !== void 0 && (t.r = n), t.S.n = t, t.i = -1, t.n === void 0) {
      e.s = t;
      break;
    }
  }
}
function Ot(e) {
  for (var t = e.s, n = void 0; t !== void 0; ) {
    var r = t.p;
    t.i === -1 ? (t.S.U(t), r !== void 0 && (r.n = t.n), t.n !== void 0 && (t.n.p = r)) : n = t, t.S.n = t.r, t.r !== void 0 && (t.r = void 0), t = r;
  }
  e.s = n;
}
function J(e) {
  S.call(this, void 0), this.x = e, this.s = void 0, this.g = he - 1, this.f = 4;
}
(J.prototype = new S()).h = function() {
  if (this.f &= -3, 1 & this.f) return !1;
  if ((36 & this.f) == 32 || (this.f &= -5, this.g === he)) return !0;
  if (this.g = he, this.f |= 1, this.i > 0 && !Ct(this))
    return this.f &= -2, !0;
  var e = y;
  try {
    Tt(this), y = this;
    var t = this.x();
    (16 & this.f || this.v !== t || this.i === 0) && (this.v = t, this.f &= -17, this.i++);
  } catch (n) {
    this.v = n, this.f |= 16, this.i++;
  }
  return y = e, Ot(this), this.f &= -2, !0;
};
J.prototype.S = function(e) {
  if (this.t === void 0) {
    this.f |= 36;
    for (var t = this.s; t !== void 0; t = t.n) t.S.S(t);
  }
  S.prototype.S.call(this, e);
};
J.prototype.U = function(e) {
  if (this.t !== void 0 && (S.prototype.U.call(this, e), this.t === void 0)) {
    this.f &= -33;
    for (var t = this.s; t !== void 0; t = t.n) t.S.U(t);
  }
};
J.prototype.N = function() {
  if (!(2 & this.f)) {
    this.f |= 6;
    for (var e = this.t; e !== void 0; e = e.x) e.t.N();
  }
};
Object.defineProperty(J.prototype, "value", { get: function() {
  if (1 & this.f) throw new Error("Cycle detected");
  var e = Et(this);
  if (this.h(), e !== void 0 && (e.i = this.i), 16 & this.f) throw this.v;
  return this.v;
} });
function ut(e) {
  return new J(e);
}
function $t(e) {
  var t = e.u;
  if (e.u = void 0, typeof t == "function") {
    j++;
    var n = y;
    y = void 0;
    try {
      t();
    } catch (r) {
      throw e.f &= -2, e.f |= 8, Ie(e), r;
    } finally {
      y = n, ke();
    }
  }
}
function Ie(e) {
  for (var t = e.s; t !== void 0; t = t.n) t.S.U(t);
  e.x = void 0, e.s = void 0, $t(e);
}
function Ln(e) {
  if (y !== this) throw new Error("Out-of-order effect");
  Ot(this), y = e, this.f &= -2, 8 & this.f && Ie(this), ke();
}
function ne(e) {
  this.x = e, this.u = void 0, this.s = void 0, this.o = void 0, this.f = 32;
}
ne.prototype.c = function() {
  var e = this.S();
  try {
    if (8 & this.f || this.x === void 0) return;
    var t = this.x();
    typeof t == "function" && (this.u = t);
  } finally {
    e();
  }
};
ne.prototype.S = function() {
  if (1 & this.f) throw new Error("Cycle detected");
  this.f |= 1, this.f &= -9, $t(this), Tt(this), j++;
  var e = y;
  return y = this, Ln.bind(this, e);
};
ne.prototype.N = function() {
  2 & this.f || (this.f |= 2, this.o = ee, ee = this);
};
ne.prototype.d = function() {
  this.f |= 8, 1 & this.f || Ie(this);
};
function H(e) {
  var t = new ne(e);
  try {
    t.c();
  } catch (n) {
    throw t.d(), n;
  }
  return t.d.bind(t);
}
var At, we, Wt = [];
H(function() {
  At = this.N;
})();
function G(e, t) {
  h[e] = t.bind(null, h[e] || function() {
  });
}
function pe(e) {
  we && we(), we = e && e.S();
}
function Ft(e) {
  var t = this, n = e.data, r = Cn(n);
  r.value = n;
  var i = St(function() {
    for (var s = t, d = t.__v; d = d.__; ) if (d.__c) {
      d.__c.__$f |= 4;
      break;
    }
    var l = ut(function() {
      var f = r.value.value;
      return f === 0 ? 0 : f === !0 ? "" : f || "";
    }), _ = ut(function() {
      return !ht(l.value);
    }), u = H(function() {
      if (this.N = jt, _.value) {
        var f = l.value;
        s.__v && s.__v.__e && s.__v.__e.nodeType === 3 && (s.__v.__e.data = f);
      }
    }), c = t.__$u.d;
    return t.__$u.d = function() {
      u(), c.call(this);
    }, [_, l];
  }, []), o = i[0], a = i[1];
  return o.value ? a.peek() : a.value;
}
Ft.displayName = "_st";
Object.defineProperties(S.prototype, { constructor: { configurable: !0, value: void 0 }, type: { configurable: !0, value: Ft }, props: { configurable: !0, get: function() {
  return { data: this };
} }, __b: { configurable: !0, value: 1 } });
G("__b", function(e, t) {
  if (typeof t.type == "string") {
    var n, r = t.props;
    for (var i in r) if (i !== "children") {
      var o = r[i];
      o instanceof S && (n || (t.__np = n = {}), n[i] = o, r[i] = o.peek());
    }
  }
  e(t);
});
G("__r", function(e, t) {
  pe();
  var n, r = t.__c;
  r && (r.__$f &= -2, (n = r.__$u) === void 0 && (r.__$u = n = function(i) {
    var o;
    return H(function() {
      o = this;
    }), o.c = function() {
      r.__$f |= 1, r.setState({});
    }, o;
  }())), pe(n), e(t);
});
G("__e", function(e, t, n, r) {
  pe(), e(t, n, r);
});
G("diffed", function(e, t) {
  pe();
  var n;
  if (typeof t.type == "string" && (n = t.__e)) {
    var r = t.__np, i = t.props;
    if (r) {
      var o = n.U;
      if (o) for (var a in o) {
        var s = o[a];
        s !== void 0 && !(a in r) && (s.d(), o[a] = void 0);
      }
      else
        o = {}, n.U = o;
      for (var d in r) {
        var l = o[d], _ = r[d];
        l === void 0 ? (l = En(n, d, _, i), o[d] = l) : l.o(_, i);
      }
    }
  }
  e(t);
});
function En(e, t, n, r) {
  var i = t in e && e.ownerSVGElement === void 0, o = M(n);
  return { o: function(a, s) {
    o.value = a, r = s;
  }, d: H(function() {
    this.N = jt;
    var a = o.value.value;
    r[t] !== a && (r[t] = a, i ? e[t] = a : a ? e.setAttribute(t, a) : e.removeAttribute(t));
  }) };
}
G("unmount", function(e, t) {
  if (typeof t.type == "string") {
    var n = t.__e;
    if (n) {
      var r = n.U;
      if (r) {
        n.U = void 0;
        for (var i in r) {
          var o = r[i];
          o && o.d();
        }
      }
    }
  } else {
    var a = t.__c;
    if (a) {
      var s = a.__$u;
      s && (a.__$u = void 0, s.d());
    }
  }
  e(t);
});
G("__h", function(e, t, n, r) {
  (r < 3 || r === 9) && (t.__$f |= 2), e(t, n, r);
});
O.prototype.shouldComponentUpdate = function(e, t) {
  var n = this.__$u, r = n && n.s !== void 0;
  for (var i in t) return !0;
  if (this.__f || typeof this.u == "boolean" && this.u === !0) {
    var o = 2 & this.__$f;
    if (!(r || o || 4 & this.__$f) || 1 & this.__$f) return !0;
  } else if (!(r || 4 & this.__$f) || 3 & this.__$f) return !0;
  for (var a in e) if (a !== "__source" && e[a] !== this.props[a]) return !0;
  for (var s in this.props) if (!(s in e)) return !0;
  return !1;
};
function Cn(e) {
  return St(function() {
    return M(e);
  }, []);
}
var Tn = function(e) {
  queueMicrotask(function() {
    queueMicrotask(e);
  });
};
function On() {
  xn(function() {
    for (var e; e = Wt.shift(); ) At.call(e);
  });
}
function jt() {
  Wt.push(this) === 1 && (h.requestAnimationFrame || Tn)(On);
}
const A = un({}), dt = {
  DATE_STRING: /^\d{4}-\d{2}-\d{2}$/,
  DATE_TIME_STRING: /^\d{4}-\d{2}-\d{2} \d{2}:\d{2}$/
};
class $n extends Error {
  constructor(t) {
    super(`Invalid date time specification: ${t}`);
  }
}
const I = (e) => {
  if (!dt.DATE_TIME_STRING.test(e) && !dt.DATE_STRING.test(e))
    throw new $n(e);
  return new Date(
    Number(e.slice(0, 4)),
    Number(e.slice(5, 7)) - 1,
    Number(e.slice(8, 10)),
    Number(e.slice(11, 13)),
    // for date strings this will be 0
    Number(e.slice(14, 16))
    // for date strings this will be 0
  );
}, re = (e) => {
  const t = e.slice(11, 13), n = e.slice(14, 16);
  return {
    year: Number(e.slice(0, 4)),
    month: Number(e.slice(5, 7)) - 1,
    date: Number(e.slice(8, 10)),
    hours: t !== "" ? Number(t) : void 0,
    minutes: n !== "" ? Number(n) : void 0
  };
}, It = (e, t) => e.toLocaleString(t, { month: "long" }), Rt = (e, t) => e.toLocaleString(t, {
  month: "numeric",
  day: "numeric",
  year: "numeric"
}), An = (e, t) => e.map((n) => n.toLocaleString(t, { weekday: "short" }).charAt(0)), Wn = (e, t) => t === "he-IL" ? e.toLocaleString(t, { weekday: "narrow" }) : e.toLocaleString(t, { weekday: "short" }), Fn = (e, t) => e.map((n) => Wn(n, t)), jn = (e, t) => ["zh-cn", "zh-tw", "ca-es", "he-il"].includes(t.toLowerCase()) ? Fn(e, t) : An(e, t);
var In = "data:image/svg+xml,%3c%3fxml version='1.0' encoding='utf-8'%3f%3e%3c!-- Uploaded to: SVG Repo%2c www.svgrepo.com%2c Generator: SVG Repo Mixer Tools --%3e%3csvg width='800px' height='800px' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3e%3cpath d='M6 9L12 15L18 9' stroke='%23DED8E1' stroke-width='4' stroke-linecap='round' stroke-linejoin='round'/%3e%3c/svg%3e";
const ce = () => "s" + Math.random().toString(36).substring(2, 11), Vt = (e) => e.key === "Enter" || e.key === " ";
function Rn() {
  const e = ce(), t = ce(), n = ce(), r = R(A), i = (c) => c === "" ? r.translate("MM/DD/YYYY") : Rt(I(c), r.config.locale.value);
  C(() => {
    r.datePickerState.inputDisplayedValue.value = i(r.datePickerState.selectedDate.value);
  }, [r.datePickerState.selectedDate.value, r.config.locale.value]);
  const [o, a] = $([]), s = () => {
    const c = document.getElementById(n);
    r.datePickerState.inputWrapperElement.value = c instanceof HTMLDivElement ? c : void 0;
  };
  C(() => {
    r.config.teleportTo && s();
    const c = ["sx__date-input-wrapper"];
    r.datePickerState.isOpen.value && c.push("sx__date-input--active"), a(c);
  }, [r.datePickerState.isOpen.value]);
  const d = (c) => {
    c.key === "Enter" && l(c);
  }, l = (c) => {
    c.stopPropagation();
    try {
      r.datePickerState.inputDisplayedValue.value = c.target.value, r.datePickerState.close();
    } catch (f) {
      console.log("Error setting input value:" + f);
    }
  };
  C(() => {
    const c = document.getElementById(e);
    if (c !== null)
      return c.addEventListener("change", l), () => c.removeEventListener("change", l);
  });
  const _ = (c) => {
    l(c), r.datePickerState.open();
  }, u = (c) => {
    Vt(c) && (c.preventDefault(), r.datePickerState.open(), setTimeout(() => {
      const f = document.querySelector('[data-focus="true"]');
      f instanceof HTMLElement && f.focus();
    }, 50));
  };
  return p(Y, { children: p("div", { className: o.join(" "), id: n, children: [p("label", { for: e, id: t, className: "sx__date-input-label", children: r.config.label || r.translate("Date") }), p("input", { id: e, tabIndex: r.datePickerState.isDisabled.value ? -1 : 0, name: r.config.name || "date", "aria-describedby": t, value: r.datePickerState.inputDisplayedValue.value, "data-testid": "date-picker-input", className: "sx__date-input", onClick: _, onKeyUp: d, type: "text" }), p("button", { type: "button", tabIndex: r.datePickerState.isDisabled.value ? -1 : 0, "aria-label": r.translate("Choose Date"), onKeyDown: u, onClick: () => r.datePickerState.open(), className: "sx__date-input-chevron-wrapper", children: p("img", { className: "sx__date-input-chevron", src: In, alt: "" }) })] }) });
}
var U;
(function(e) {
  e.MONTH_DAYS = "month-days", e.YEARS = "years";
})(U || (U = {}));
const Vn = "years-view", Un = "months-view", Hn = "date-picker-week";
class zn extends Error {
  constructor(t, n) {
    super(`Number must be between ${t} and ${n}.`), Object.defineProperty(this, "min", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: t
    }), Object.defineProperty(this, "max", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: n
    });
  }
}
const N = (e) => {
  if (e < 0 || e > 99)
    throw new zn(0, 99);
  return String(e).padStart(2, "0");
}, E = (e) => `${e.getFullYear()}-${N(e.getMonth() + 1)}-${N(e.getDate())}`, Kn = (e) => `${N(e.getHours())}:${N(e.getMinutes())}`, Ut = (e) => `${E(e)} ${Kn(e)}`, Ht = (e, t) => {
  const { year: n, month: r, date: i, hours: o, minutes: a } = re(e), s = o !== void 0 && a !== void 0, d = new Date(n, r, i, o ?? 0, a ?? 0);
  let l = (d.getMonth() + t) % 12;
  return l < 0 && (l += 12), d.setMonth(d.getMonth() + t), d.getMonth() > l ? d.setDate(0) : d.getMonth() < l && (d.setMonth(d.getMonth() + 1), d.setDate(0)), s ? Ut(d) : E(d);
}, Bn = (e, t) => {
  const { year: n, month: r, date: i, hours: o, minutes: a } = re(e), s = o !== void 0 && a !== void 0, d = new Date(n, r, i, o ?? 0, a ?? 0);
  return d.setDate(d.getDate() + t), s ? Ut(d) : E(d);
}, zt = (e, t) => (e = e.slice(0, 8) + N(t) + e.slice(10), e), Jn = (e) => (e = Ht(e, -1), zt(e, 1)), Gn = (e) => (e = Ht(e, 1), zt(e, 1));
function lt({ direction: e, onClick: t, buttonText: n, disabled: r = !1 }) {
  return p("button", { type: "button", disabled: r, className: "sx__chevron-wrapper sx__ripple", onMouseUp: t, onKeyDown: (o) => {
    Vt(o) && t();
  }, tabIndex: 0, children: p("i", { className: `sx__chevron sx__chevron--${e}`, children: n }) });
}
function Zn({ setYearsView: e }) {
  const t = R(A), n = (u) => {
    const c = I(u);
    return It(c, t.config.locale.value);
  }, r = (u) => re(u).year, [i, o] = $(n(t.datePickerState.datePickerDate.value)), [a, s] = $(r(t.datePickerState.datePickerDate.value)), d = () => {
    t.datePickerState.datePickerDate.value = Jn(t.datePickerState.datePickerDate.value);
  }, l = () => {
    t.datePickerState.datePickerDate.value = Gn(t.datePickerState.datePickerDate.value);
  };
  C(() => {
    o(n(t.datePickerState.datePickerDate.value)), s(r(t.datePickerState.datePickerDate.value));
  }, [t.datePickerState.datePickerDate.value]);
  const _ = (u) => {
    u.stopPropagation(), e();
  };
  return p(Y, { children: p("header", { className: "sx__date-picker__month-view-header", children: [p(lt, { direction: "previous", onClick: () => d(), buttonText: t.translate("Previous month") }), p("button", { type: "button", className: "sx__date-picker__month-view-header__month-year", onClick: (u) => _(u), children: i + " " + a }), p(lt, { direction: "next", onClick: () => l(), buttonText: t.translate("Next month") })] }) });
}
function qn() {
  const e = R(A), t = e.timeUnitsImpl.getWeekFor(I(e.datePickerState.datePickerDate.value)), n = jn(t, e.config.locale.value);
  return p("div", { className: "sx__date-picker__day-names", children: n.map((r) => p("span", { "data-testid": "day-name", className: "sx__date-picker__day-name", children: r })) });
}
const Xn = (e) => {
  const t = /* @__PURE__ */ new Date();
  return e.getDate() === t.getDate() && e.getMonth() === t.getMonth() && e.getFullYear() === t.getFullYear();
}, Qn = (e, t) => e.getMonth() === t.getMonth() && e.getFullYear() === t.getFullYear();
var Te;
(function(e) {
  e[e.SUNDAY = 0] = "SUNDAY", e[e.MONDAY = 1] = "MONDAY", e[e.TUESDAY = 2] = "TUESDAY", e[e.WEDNESDAY = 3] = "WEDNESDAY", e[e.THURSDAY = 4] = "THURSDAY", e[e.FRIDAY = 5] = "FRIDAY", e[e.SATURDAY = 6] = "SATURDAY";
})(Te || (Te = {}));
const er = "en-US", tr = Te.MONDAY, nr = (e, t) => {
  const { year: n, month: r, date: i } = re(e);
  return new Date(n, r, i).toLocaleDateString(t, {
    day: "numeric",
    month: "long",
    year: "numeric"
  });
}, rr = nr;
function ir({ week: e }) {
  const t = R(A), n = e.map((s) => {
    const d = ["sx__date-picker__day"];
    return Xn(s) && d.push("sx__date-picker__day--today"), E(s) === t.datePickerState.selectedDate.value && d.push("sx__date-picker__day--selected"), Qn(s, I(t.datePickerState.datePickerDate.value)) || d.push("is-leading-or-trailing"), {
      day: s,
      classes: d
    };
  }), r = (s) => {
    const d = E(s);
    return d >= t.config.min && d <= t.config.max;
  }, i = (s) => {
    t.datePickerState.selectedDate.value = E(s), t.datePickerState.close();
  }, o = (s) => E(s.day) === t.datePickerState.datePickerDate.value, a = (s) => {
    if (s.key === "Enter") {
      t.datePickerState.selectedDate.value = t.datePickerState.datePickerDate.value, t.datePickerState.close();
      return;
    }
    const d = /* @__PURE__ */ new Map([
      ["ArrowDown", 7],
      ["ArrowUp", -7],
      ["ArrowLeft", -1],
      ["ArrowRight", 1]
    ]);
    t.datePickerState.datePickerDate.value = Bn(t.datePickerState.datePickerDate.value, d.get(s.key) || 0);
  };
  return p(Y, { children: p("div", { "data-testid": Hn, className: "sx__date-picker__week", children: n.map((s) => p("button", { type: "button", tabIndex: o(s) ? 0 : -1, disabled: !r(s.day), "aria-label": rr(t.datePickerState.datePickerDate.value, t.config.locale.value), className: s.classes.join(" "), "data-focus": o(s) ? "true" : void 0, onClick: () => i(s.day), onKeyDown: a, children: s.day.getDate() })) }) });
}
function or({ seatYearsView: e }) {
  const t = ce(), n = R(A), [r, i] = $([]), o = () => {
    const a = I(n.datePickerState.datePickerDate.value);
    i(n.timeUnitsImpl.getMonthWithTrailingAndLeadingDays(a.getFullYear(), a.getMonth()));
  };
  return C(() => {
    o();
  }, [n.datePickerState.datePickerDate.value]), C(() => {
    const a = new MutationObserver((d) => {
      d.forEach((l) => {
        const _ = l.target;
        _.dataset.focus === "true" && _.focus();
      });
    }), s = document.getElementById(t);
    return a.observe(s, {
      childList: !0,
      subtree: !0,
      attributes: !0
    }), () => a.disconnect();
  }, []), p(Y, { children: p("div", { id: t, "data-testid": Un, className: "sx__date-picker__month-view", children: [p(Zn, { setYearsView: e }), p(qn, {}), r.map((a) => p(ir, { week: a }))] }) });
}
function ar({ year: e, setYearAndMonth: t, isExpanded: n, expand: r }) {
  const i = R(A), o = i.timeUnitsImpl.getMonthsFor(e), a = (s, d) => {
    s.stopPropagation(), t(e, d.getMonth());
  };
  return p(Y, { children: p("li", { className: n ? "sx__is-expanded" : "", children: [p("button", { type: "button", className: "sx__date-picker__years-accordion__expand-button sx__ripple--wide", onClick: () => r(e), children: e }), n && p("div", { className: "sx__date-picker__years-view-accordion__panel", children: o.map((s) => p("button", { type: "button", className: "sx__date-picker__years-view-accordion__month", onClick: (d) => a(d, s), children: It(s, i.config.locale.value) })) })] }) });
}
function sr({ setMonthView: e }) {
  const t = R(A), n = I(t.config.min).getFullYear(), r = I(t.config.max).getFullYear(), i = Array.from({ length: r - n + 1 }, (l, _) => n + _), { year: o } = re(t.datePickerState.selectedDate.value), [a, s] = $(o), d = (l, _) => {
    t.datePickerState.datePickerDate.value = E(new Date(l, _, 1)), e();
  };
  return C(() => {
    var l;
    const _ = (l = document.querySelector(".sx__date-picker__years-view")) === null || l === void 0 ? void 0 : l.querySelector(".sx__is-expanded");
    _ && _.scrollIntoView({
      block: "center"
    });
  }, []), p(Y, { children: p("ul", { className: "sx__date-picker__years-view", "data-testid": Vn, children: i.map((l) => p(ar, { year: l, setYearAndMonth: (_, u) => d(_, u), isExpanded: a === l, expand: (_) => s(_) })) }) });
}
const ur = (e) => {
  if (e) {
    const t = e.scrollHeight > e.clientHeight, r = window.getComputedStyle(e).overflowY.indexOf("hidden") !== -1;
    return t && !r;
  }
  return !0;
}, Kt = (e, t = []) => !e || e === document.body || e.nodeType === Node.DOCUMENT_FRAGMENT_NODE ? (t.push(window), t) : (ur(e) && t.push(e), Kt(e.assignedSlot ? e.assignedSlot.parentNode : e.parentNode, t)), ct = "sx__date-picker-popup";
function dr() {
  const e = R(A), [t, n] = $(U.MONTH_DAYS), r = [ct, e.config.placement], [i, o] = $(r);
  C(() => {
    o([
      ...r,
      e.datePickerState.isDark.value ? "is-dark" : "",
      e.config.teleportTo ? "is-teleported" : ""
    ]);
  }, [e.datePickerState.isDark.value]);
  const a = (m) => {
    m.target.closest(`.${ct}`) || e.datePickerState.close();
  }, s = (m) => {
    m.key === "Escape" && (e.config.listeners.onEscapeKeyDown ? e.config.listeners.onEscapeKeyDown(e) : e.datePickerState.close());
  };
  C(() => (document.addEventListener("click", a), document.addEventListener("keydown", s), () => {
    document.removeEventListener("click", a), document.removeEventListener("keydown", s);
  }), []);
  const d = Number(getComputedStyle(document.documentElement).fontSize.split("px")[0]), l = 362, _ = 332, u = () => {
    const m = e.datePickerState.inputWrapperElement.value, v = m == null ? void 0 : m.getBoundingClientRect();
    if (!(m === void 0 || !(v instanceof DOMRect)))
      return {
        top: e.config.placement.includes("bottom") ? v.height + v.y + 1 : v.y - d - l,
        // subtract remsize to leave room for label text
        left: e.config.placement.includes("start") ? v.x : v.x + v.width - _,
        width: _,
        position: "fixed"
      };
  }, [c, f] = $(u());
  return C(() => {
    const m = e.datePickerState.inputWrapperElement.value;
    if (m === void 0)
      return;
    const v = Kt(m), P = () => f(u());
    return v.forEach((k) => k.addEventListener("scroll", P)), () => v.forEach((k) => k.removeEventListener("scroll", P));
  }, []), p(Y, { children: p("div", { style: e.config.teleportTo ? c : void 0, "data-testid": "date-picker-popup", className: i.join(" "), children: t === U.MONTH_DAYS ? p(or, { seatYearsView: () => n(U.YEARS) }) : p(sr, { setMonthView: () => n(U.MONTH_DAYS) }) }) });
}
function lr({ $app: e }) {
  const t = ["sx__date-picker-wrapper"], [n, r] = $(t);
  C(() => {
    var o;
    const a = [...t];
    e.datePickerState.isDark.value && a.push("is-dark"), !((o = e.config.style) === null || o === void 0) && o.fullWidth && a.push("has-full-width"), e.datePickerState.isDisabled.value && a.push("is-disabled"), r(a);
  }, [e.datePickerState.isDark.value, e.datePickerState.isDisabled.value]);
  let i = p(dr, {});
  return e.config.teleportTo && (i = mn(i, e.config.teleportTo)), p(Y, { children: p("div", { className: n.join(" "), children: p(A.Provider, { value: e, children: [p(Rn, {}), e.datePickerState.isOpen.value && i] }) }) });
}
class cr {
  constructor(t) {
    Object.defineProperty(this, "$app", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: t
    });
  }
  render(t) {
    xe(z(lr, {
      $app: this.$app
    }), t);
  }
  get value() {
    return this.$app.datePickerState.selectedDate.value;
  }
  set value(t) {
    this.$app.datePickerState.selectedDate.value = t;
  }
  get disabled() {
    return this.$app.datePickerState.isDisabled.value;
  }
  set disabled(t) {
    this.$app.datePickerState.isDisabled.value = t;
  }
  setTheme(t) {
    this.$app.datePickerState.isDark.value = t === "dark";
  }
  getTheme() {
    return this.$app.datePickerState.isDark.value ? "dark" : "light";
  }
}
var Oe;
(function(e) {
  e[e.JANUARY = 0] = "JANUARY", e[e.FEBRUARY = 1] = "FEBRUARY", e[e.MARCH = 2] = "MARCH", e[e.APRIL = 3] = "APRIL", e[e.MAY = 4] = "MAY", e[e.JUNE = 5] = "JUNE", e[e.JULY = 6] = "JULY", e[e.AUGUST = 7] = "AUGUST", e[e.SEPTEMBER = 8] = "SEPTEMBER", e[e.OCTOBER = 9] = "OCTOBER", e[e.NOVEMBER = 10] = "NOVEMBER", e[e.DECEMBER = 11] = "DECEMBER";
})(Oe || (Oe = {}));
class Bt extends Error {
  constructor() {
    super("Year zero does not exist in the Gregorian calendar.");
  }
}
class _r extends Date {
  constructor(t, n, r) {
    if (super(t, n, r), t === 0)
      throw new Bt();
    this.setFullYear(t);
  }
  get year() {
    return this.getFullYear();
  }
  get month() {
    return this.getMonth();
  }
  get date() {
    return this.getDate();
  }
}
class fr {
  constructor(t) {
    Object.defineProperty(this, "config", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: t
    });
  }
  get firstDayOfWeek() {
    return this.config.firstDayOfWeek.value;
  }
  set firstDayOfWeek(t) {
    this.config.firstDayOfWeek.value = t;
  }
  getMonthWithTrailingAndLeadingDays(t, n) {
    if (t === 0)
      throw new Bt();
    const r = new Date(t, n, 1), i = [this.getWeekFor(r)];
    let o = !0, a = i[0][0];
    for (; o; ) {
      const s = new Date(a.getFullYear(), a.getMonth(), a.getDate() + 7);
      s.getMonth() === n ? (i.push(this.getWeekFor(s)), a = s) : o = !1;
    }
    return i;
  }
  getWeekFor(t) {
    const n = [this.getFirstDateOfWeek(t)];
    for (; n.length < 7; ) {
      const r = n[n.length - 1], i = new Date(r);
      i.setDate(r.getDate() + 1), n.push(i);
    }
    return n;
  }
  getMonthsFor(t) {
    return Object.values(Oe).filter((n) => !isNaN(Number(n))).map((n) => new _r(t, Number(n), 1));
  }
  getFirstDateOfWeek(t) {
    const n = t.getDay() - this.firstDayOfWeek, r = t;
    return n === 0 || (n > 0 ? r.setDate(t.getDate() - n) : r.setDate(t.getDate() - (7 + n))), r;
  }
}
class hr {
  constructor() {
    Object.defineProperty(this, "config", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    });
  }
  build() {
    return new fr(this.config);
  }
  withConfig(t) {
    return this.config = t, this;
  }
}
var x;
(function(e) {
  e.SLASH = "/", e.DASH = "-", e.PERIOD = ".";
})(x || (x = {}));
var L;
(function(e) {
  e.DMY = "DMY", e.MDY = "MDY", e.YMD = "YMD";
})(L || (L = {}));
const D = {
  slashMDY: {
    delimiter: x.SLASH,
    order: L.MDY
  },
  slashDMY: {
    delimiter: x.SLASH,
    order: L.DMY
  },
  slashYMD: {
    delimiter: x.SLASH,
    order: L.YMD
  },
  periodDMY: {
    delimiter: x.PERIOD,
    order: L.DMY
  },
  dashYMD: {
    delimiter: x.DASH,
    order: L.YMD
  },
  dashDMY: {
    delimiter: x.DASH,
    order: L.DMY
  }
}, pr = /* @__PURE__ */ new Map([
  ["ca-ES", D.slashDMY],
  ["cs-CZ", D.periodDMY],
  ["da-DK", D.periodDMY],
  ["de-DE", D.periodDMY],
  ["en-GB", D.slashDMY],
  ["en-US", D.slashMDY],
  ["es-ES", D.slashDMY],
  ["et-EE", D.periodDMY],
  ["fi-FI", D.periodDMY],
  ["fr-FR", D.slashDMY],
  ["hr-HR", D.periodDMY],
  ["id-ID", D.slashDMY],
  ["it-IT", D.slashDMY],
  ["ja-JP", D.slashYMD],
  ["ko-KR", D.slashYMD],
  ["ky-KG", D.slashDMY],
  ["lt-LT", D.dashYMD],
  ["mk-MK", D.periodDMY],
  ["nl-NL", D.dashDMY],
  ["pl-PL", D.periodDMY],
  ["pt-BR", D.slashDMY],
  ["ro-RO", D.periodDMY],
  ["ru-RU", D.periodDMY],
  ["sk-SK", D.periodDMY],
  ["sl-SI", D.periodDMY],
  ["sr-Latn-RS", D.periodDMY],
  ["sr-RS", D.periodDMY],
  ["sv-SE", D.dashYMD],
  ["tr-TR", D.periodDMY],
  ["uk-UA", D.periodDMY],
  ["zh-CN", D.slashYMD],
  ["zh-TW", D.slashYMD]
]);
class vr extends Error {
  constructor(t) {
    super(`Locale not supported: ${t}`);
  }
}
class Jt extends Error {
  constructor(t, n) {
    super(`Invalid date format: ${t} for locale: ${n}`);
  }
}
const ue = (e, t, n) => {
  const r = e.match(t);
  if (!r)
    throw new Jt(e, n);
  return r;
}, mr = (e, t) => {
  if (/^\d{4}-\d{2}-\d{2}$/.test(e))
    return e;
  const r = pr.get(t);
  if (!r)
    throw new vr(t);
  const { order: i, delimiter: o } = r, a = /^(\d{1,2})\/(\d{1,2})\/(\d{4})$/, s = /^(\d{1,2})\.(\d{1,2})\.(\d{4})$/, d = /^(\d{4})\/(\d{1,2})\/(\d{1,2})$/;
  if (i === L.DMY && o === x.SLASH) {
    const l = ue(e, a, t), [, _, u, c] = l;
    return `${c}-${N(+u)}-${N(+_)}`;
  }
  if (i === L.MDY && o === x.SLASH) {
    const l = ue(e, a, t), [, _, u, c] = l;
    return `${c}-${N(+_)}-${N(+u)}`;
  }
  if (i === L.YMD && o === x.SLASH) {
    const l = ue(e, d, t), [, _, u, c] = l;
    return `${_}-${N(+u)}-${N(+c)}`;
  }
  if (i === L.DMY && o === x.PERIOD) {
    const l = ue(e, s, t), [, _, u, c] = l;
    return `${c}-${N(+u)}-${N(+_)}`;
  }
  throw new Jt(e, t);
}, Dr = (e, t) => {
  var n;
  const r = E(/* @__PURE__ */ new Date()), i = typeof t == "string" ? t : r, o = M(!1), a = M(e.disabled || !1), s = M(U.MONTH_DAYS), d = M(i), l = M(i || r), _ = M(((n = e.style) === null || n === void 0 ? void 0 : n.dark) || !1), u = M(t ? Rt(I(t), e.locale.value) : ""), c = M(u.value);
  H(() => {
    try {
      const v = mr(u.value, e.locale.value);
      if (v < e.min || v > e.max) {
        u.value = c.value;
        return;
      }
      d.value = v, l.value = v, c.value = u.value;
    } catch {
    }
  });
  let f = !1;
  const m = (v) => {
    if (!f)
      return f = !0;
    e.listeners.onChange(v);
  };
  return H(() => {
    var v;
    !((v = e.listeners) === null || v === void 0) && v.onChange && m(d.value);
  }), {
    inputWrapperElement: M(void 0),
    isOpen: o,
    isDisabled: a,
    datePickerView: s,
    selectedDate: d,
    datePickerDate: l,
    inputDisplayedValue: u,
    isDark: _,
    open: () => o.value = !0,
    close: () => o.value = !1,
    toggle: () => o.value = !o.value,
    setView: (v) => s.value = v
  };
};
class kr {
  constructor(t, n, r, i) {
    Object.defineProperty(this, "datePickerState", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: t
    }), Object.defineProperty(this, "config", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: n
    }), Object.defineProperty(this, "timeUnitsImpl", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: r
    }), Object.defineProperty(this, "translate", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: i
    });
  }
}
class yr {
  constructor() {
    Object.defineProperty(this, "datePickerState", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "config", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "timeUnitsImpl", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "translate", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    });
  }
  build() {
    return new kr(this.datePickerState, this.config, this.timeUnitsImpl, this.translate);
  }
  withDatePickerState(t) {
    return this.datePickerState = t, this;
  }
  withConfig(t) {
    return this.config = t, this;
  }
  withTimeUnitsImpl(t) {
    return this.timeUnitsImpl = t, this;
  }
  withTranslate(t) {
    return this.translate = t, this;
  }
}
var $e;
(function(e) {
  e.TOP_START = "top-start", e.TOP_END = "top-end", e.BOTTOM_START = "bottom-start", e.BOTTOM_END = "bottom-end";
})($e || ($e = {}));
class gr {
  constructor(t = er, n = tr, r = E(new Date(1970, 0, 1)), i = E(new Date((/* @__PURE__ */ new Date()).getFullYear() + 50, 11, 31)), o = $e.BOTTOM_START, a = {}, s = {}, d, l, _, u) {
    Object.defineProperty(this, "min", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: r
    }), Object.defineProperty(this, "max", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: i
    }), Object.defineProperty(this, "placement", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: o
    }), Object.defineProperty(this, "listeners", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: a
    }), Object.defineProperty(this, "style", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: s
    }), Object.defineProperty(this, "teleportTo", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: d
    }), Object.defineProperty(this, "label", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: l
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: _
    }), Object.defineProperty(this, "disabled", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: u
    }), Object.defineProperty(this, "locale", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "firstDayOfWeek", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), this.locale = M(t), this.firstDayOfWeek = M(n);
  }
}
class br {
  constructor() {
    Object.defineProperty(this, "locale", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "firstDayOfWeek", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "min", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "max", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "placement", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "listeners", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "style", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "teleportTo", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "label", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "disabled", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    });
  }
  build() {
    return new gr(this.locale, this.firstDayOfWeek, this.min, this.max, this.placement, this.listeners, this.style, this.teleportTo, this.label, this.name, this.disabled);
  }
  withLocale(t) {
    return this.locale = t, this;
  }
  withFirstDayOfWeek(t) {
    return this.firstDayOfWeek = t, this;
  }
  withMin(t) {
    return this.min = t, this;
  }
  withMax(t) {
    return this.max = t, this;
  }
  withPlacement(t) {
    return this.placement = t, this;
  }
  withListeners(t) {
    return this.listeners = t, this;
  }
  withStyle(t) {
    return this.style = t, this;
  }
  withTeleportTo(t) {
    return this.teleportTo = t, this;
  }
  withLabel(t) {
    return this.label = t, this;
  }
  withName(t) {
    return this.name = t, this;
  }
  withDisabled(t) {
    return this.disabled = t, this;
  }
}
const Pr = {
  Date: "Datum",
  "MM/DD/YYYY": "TT.MM.JJJJ",
  "Next month": "Nchster Monat",
  "Previous month": "Vorheriger Monat",
  "Choose Date": "Datum auswhlen"
}, wr = {
  Today: "Heute",
  Month: "Monat",
  Week: "Woche",
  Day: "Tag",
  "Select View": "Ansicht auswhlen",
  events: "Ereignisse",
  event: "Ereignis",
  "No events": "Keine Ereignisse",
  "Next period": "Nchster Zeitraum",
  "Previous period": "Vorheriger Zeitraum",
  to: "bis",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Ganztgige und mehrtgige Ereignisse",
  "Link to {{n}} more events on {{date}}": "Link zu {{n}} weiteren Ereignissen am {{date}}",
  "Link to 1 more event on {{date}}": "Link zu 1 weiteren Ereignis am {{date}}",
  CW: "KW {{week}}"
}, Sr = {
  ...Pr,
  ...wr
}, Mr = {
  Date: "Date",
  "MM/DD/YYYY": "MM/DD/YYYY",
  "Next month": "Next month",
  "Previous month": "Previous month",
  "Choose Date": "Choose Date"
}, Yr = {
  Today: "Today",
  Month: "Month",
  Week: "Week",
  Day: "Day",
  "Select View": "Select View",
  events: "events",
  event: "event",
  "No events": "No events",
  "Next period": "Next period",
  "Previous period": "Previous period",
  to: "to",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Full day- and multiple day events",
  "Link to {{n}} more events on {{date}}": "Link to {{n}} more events on {{date}}",
  "Link to 1 more event on {{date}}": "Link to 1 more event on {{date}}",
  CW: "Week {{week}}"
}, Nr = {
  ...Mr,
  ...Yr
}, xr = {
  Date: "Data",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Mese successivo",
  "Previous month": "Mese precedente",
  "Choose Date": "Scegli la data"
}, Lr = {
  Today: "Oggi",
  Month: "Mese",
  Week: "Settimana",
  Day: "Giorno",
  "Select View": "Seleziona la vista",
  events: "eventi",
  event: "evento",
  "No events": "Nessun evento",
  "Next period": "Periodo successivo",
  "Previous period": "Periodo precedente",
  to: "a",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Eventi della giornata e plurigiornalieri",
  "Link to {{n}} more events on {{date}}": "Link a {{n}} eventi in pi il {{date}}",
  "Link to 1 more event on {{date}}": "Link a 1 evento in pi il {{date}}",
  CW: "Settimana {{week}}"
}, Er = {
  ...xr,
  ...Lr
}, Cr = {
  Date: "Date",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Next month",
  "Previous month": "Previous month",
  "Choose Date": "Choose Date"
}, Tr = {
  Today: "Today",
  Month: "Month",
  Week: "Week",
  Day: "Day",
  "Select View": "Select View",
  events: "events",
  event: "event",
  "No events": "No events",
  "Next period": "Next period",
  "Previous period": "Previous period",
  to: "to",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Full day- and multiple day events",
  "Link to {{n}} more events on {{date}}": "Link to {{n}} more events on {{date}}",
  "Link to 1 more event on {{date}}": "Link to 1 more event on {{date}}",
  CW: "Week {{week}}"
}, Or = {
  ...Cr,
  ...Tr
}, $r = {
  Date: "Datum",
  "MM/DD/YYYY": "-MM-DD",
  "Next month": "Nsta mnad",
  "Previous month": "Fregende mnad",
  "Choose Date": "Vlj datum"
}, Ar = {
  Today: "Idag",
  Month: "Mnad",
  Week: "Vecka",
  Day: "Dag",
  "Select View": "Vlj vy",
  events: "hndelser",
  event: "hndelse",
  "No events": "Inga hndelser",
  "Next period": "Nsta period",
  "Previous period": "Fregende period",
  to: "till",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Heldags- och flerdagshndelser",
  "Link to {{n}} more events on {{date}}": "Lnk till {{n}} fler hndelser den {{date}}",
  "Link to 1 more event on {{date}}": "Lnk till 1 hndelse till den {{date}}",
  CW: "Vecka {{week}}"
}, Wr = {
  ...$r,
  ...Ar
}, Fr = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": "",
  "Previous month": "",
  "Choose Date": ""
}, jr = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": "",
  events: "",
  event: "",
  "No events": "",
  "Next period": "",
  "Previous period": "",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "",
  "Link to {{n}} more events on {{date}}": "{{date}}{{n}}",
  "Link to 1 more event on {{date}}": "{{date}}1",
  CW: "{{week}}"
}, Ir = {
  ...Fr,
  ...jr
}, Rr = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": "",
  "Previous month": "",
  "Choose Date": ""
}, Vr = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": "",
  events: "",
  event: "",
  "No events": "",
  "Next period": "",
  "Previous period": "",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "",
  "Link to {{n}} more events on {{date}}": "{{date}}{{n}}",
  "Link to 1 more event on {{date}}": "{{date}}1",
  CW: "{{week}}"
}, Ur = {
  ...Rr,
  ...Vr
}, Hr = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": "",
  "Previous month": "",
  "Choose Date": ""
}, zr = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": "",
  events: "",
  event: "",
  "No events": "",
  "Next period": "",
  "Previous period": "",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "",
  "Link to {{n}} more events on {{date}}": "{{date}} {{n}}",
  "Link to 1 more event on {{date}}": "{{date}} 1",
  CW: " {{week}}"
}, Kr = {
  ...Hr,
  ...zr
}, Br = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, Jr = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "       ",
  "Link to {{n}} more events on {{date}}": "  {{n}}    {{date}}",
  "Link to 1 more event on {{date}}": "  1    {{date}}",
  CW: " {{week}}"
}, Gr = {
  ...Br,
  ...Jr
}, Zr = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, qr = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": "",
  "Previous period": "",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "   ",
  "Link to {{n}} more events on {{date}}": "{{date}} {{n}}   ",
  "Link to 1 more event on {{date}}": "{{date}} 1   ",
  CW: "{{week}}"
}, Xr = {
  ...Zr,
  ...qr
}, Qr = {
  Date: "Date",
  "MM/DD/YYYY": "JJ/MM/AAAA",
  "Next month": "Mois suivant",
  "Previous month": "Mois prcdent",
  "Choose Date": "Choisir une date"
}, ei = {
  Today: "Aujourd'hui",
  Month: "Mois",
  Week: "Semaine",
  Day: "Jour",
  "Select View": "Choisir la vue",
  events: "vnements",
  event: "vnement",
  "No events": "Aucun vnement",
  "Next period": "Priode suivante",
  "Previous period": "Priode prcdente",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "vnements d'une ou plusieurs journes",
  "Link to {{n}} more events on {{date}}": "Lien vers {{n}} autres vnements le {{date}}",
  "Link to 1 more event on {{date}}": "Lien vers 1 autre vnement le {{date}}",
  CW: "Semaine {{week}}"
}, ti = {
  ...Qr,
  ...ei
}, ni = {
  Date: "Dato",
  "MM/DD/YYYY": "-MM-DD",
  "Next month": "Nste mned",
  "Previous month": "Foregende mned",
  "Choose Date": "Vlg dato"
}, ri = {
  Today: "I dag",
  Month: "Mned",
  Week: "Uge",
  Day: "Dag",
  "Select View": "Vlg visning",
  events: "begivenheder",
  event: "begivenhed",
  "No events": "Ingen begivenheder",
  "Next period": "Nste periode",
  "Previous period": "Forgende periode",
  to: "til",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Heldagsbegivenheder og flerdagsbegivenheder",
  "Link to {{n}} more events on {{date}}": "Link til {{n}} flere begivenheder den {{date}}",
  "Link to 1 more event on {{date}}": "Link til 1 mere begivenhed den {{date}}",
  CW: "Uge {{week}}"
}, ii = {
  ...ni,
  ...ri
}, oi = {
  Date: "Data",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Nastpny miesic",
  "Previous month": "Poprzedni miesic",
  "Choose Date": "Wybiewrz dat"
}, ai = {
  Today: "Dzisiaj",
  Month: "Miesic",
  Week: "Tydzie",
  Day: "Dzie",
  "Select View": "Wybierz widok",
  events: "wydarzenia",
  event: "wydarzenie",
  "No events": "Brak wydarze",
  "Next period": "Nastpny okres",
  "Previous period": "Poprzedni okres",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Wydarzenia caodniowe i wielodniowe",
  "Link to {{n}} more events on {{date}}": "Link do {{n}} kolejnych wydarze w dniu {{date}}",
  "Link to 1 more event on {{date}}": "Link do 1 kolejnego wydarzenia w dniu {{date}}",
  CW: "Tydzie {{week}}"
}, si = {
  ...oi,
  ...ai
}, ui = {
  Date: "Fecha",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Siguiente mes",
  "Previous month": "Mes anterior",
  "Choose Date": "Seleccione una fecha"
}, di = {
  Today: "Hoy",
  Month: "Mes",
  Week: "Semana",
  Day: "Da",
  "Select View": "Seleccione una vista",
  events: "eventos",
  event: "evento",
  "No events": "Sin eventos",
  "Next period": "Siguiente perodo",
  "Previous period": "Perodo anterior",
  to: "a",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Da completo y eventos de mltiples das",
  "Link to {{n}} more events on {{date}}": "Enlace a {{n}} eventos ms el {{date}}",
  "Link to 1 more event on {{date}}": "Enlace a 1 evento ms el {{date}}",
  CW: "Semana {{week}}"
}, li = {
  ...ui,
  ...di
}, ci = {
  Today: "Vandaag",
  Month: "Maand",
  Week: "Week",
  Day: "Dag",
  "Select View": "Kies weergave",
  events: "gebeurtenissen",
  event: "gebeurtenis",
  "No events": "Geen gebeurtenissen",
  "Next period": "Volgende periode",
  "Previous period": "Vorige periode",
  to: "tot",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Evenementen van een hele dag en meerdere dagen",
  "Link to {{n}} more events on {{date}}": "Link naar {{n}} meer evenementen op {{date}}",
  "Link to 1 more event on {{date}}": "Link naar 1 meer evenement op {{date}}",
  CW: "Week {{week}}"
}, _i = {
  Date: "Datum",
  "MM/DD/YYYY": "DD-MM-JJJJ",
  "Next month": "Volgende maand",
  "Previous month": "Vorige maand",
  "Choose Date": "Kies datum"
}, fi = {
  ..._i,
  ...ci
}, hi = {
  Date: "Data",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Ms seguinte",
  "Previous month": "Ms anterior",
  "Choose Date": "Escolha uma data"
}, pi = {
  Today: "Hoje",
  Month: "Ms",
  Week: "Semana",
  Day: "Dia",
  "Select View": "Selecione uma visualizao",
  events: "eventos",
  event: "evento",
  "No events": "Sem eventos",
  "Next period": "Perodo seguinte",
  "Previous period": "Perodo anterior",
  to: "a",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Dia inteiro e eventos de vrios dias",
  "Link to {{n}} more events on {{date}}": "Link para mais {{n}} eventos em {{date}}",
  "Link to 1 more event on {{date}}": "Link para mais 1 evento em {{date}}",
  CW: "Semana {{week}}"
}, vi = {
  ...hi,
  ...pi
}, mi = {
  Date: "Dtum",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "al mesiac",
  "Previous month": "Predchdzajci mesiac",
  "Choose Date": "Vyberte dtum"
}, Di = {
  Today: "Dnes",
  Month: "Mesiac",
  Week: "Tde",
  Day: "De",
  "Select View": "Vyberte zobrazenie",
  events: "udalosti",
  event: "udalos",
  "No events": "iadne udalosti",
  "Next period": "alie obdobie",
  "Previous period": "Predchdzajce obdobie",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Celodenn a viacdov udalosti",
  "Link to {{n}} more events on {{date}}": "Odkaz na {{n}} alch udalost da {{date}}",
  "Link to 1 more event on {{date}}": "Odkaz na 1 aliu udalos da {{date}}",
  CW: "{{week}}. tde"
}, ki = {
  ...mi,
  ...Di
}, yi = {
  Date: "",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, gi = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "   ",
  "Link to {{n}} more events on {{date}}": "  {{n}}    {{date}}",
  "Link to 1 more event on {{date}}": "  1    {{date}}",
  CW: " {{week}}"
}, bi = {
  ...yi,
  ...gi
}, Pi = {
  Date: "Tarih",
  "MM/DD/YYYY": "GG/AA/YYYY",
  "Next month": "Sonraki ay",
  "Previous month": "nceki ay",
  "Choose Date": "Tarih Se"
}, wi = {
  Today: "Bugn",
  Month: "Aylk",
  Week: "Haftalk",
  Day: "Gnlk",
  "Select View": "Grnm Se",
  events: "etkinlikler",
  event: "etkinlik",
  "No events": "Etkinlik yok",
  "Next period": "Sonraki dnem",
  "Previous period": "nceki dnem",
  to: "dan",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Tm gn ve oklu gn etkinlikleri",
  "Link to {{n}} more events on {{date}}": "{{date}} tarihinde {{n}} etkinlie balant",
  "Link to 1 more event on {{date}}": "{{date}} tarihinde 1 etkinlie balant",
  CW: "{{week}}. Hafta"
}, Si = {
  ...Pi,
  ...wi
}, Mi = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, Yi = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "         ",
  "Link to {{n}} more events on {{date}}": "{{date}}  {{n}}  ",
  "Link to 1 more event on {{date}}": "{{date}}  1  ",
  CW: " {{week}}"
}, Ni = {
  ...Mi,
  ...Yi
}, xi = {
  Date: "Tanggal",
  "MM/DD/YYYY": "DD.MM.YYYY",
  "Next month": "Bulan depan",
  "Previous month": "Bulan sebelumnya",
  "Choose Date": "Pilih tanggal"
}, Li = {
  Today: "Hari Ini",
  Month: "Bulan",
  Week: "Minggu",
  Day: "Hari",
  "Select View": "Pilih tampilan",
  events: "Acara",
  event: "Acara",
  "No events": "Tidak ada acara",
  "Next period": "Periode selanjutnya",
  "Previous period": "Periode sebelumnya",
  to: "sampai",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Sepanjang hari dan acara beberapa hari ",
  "Link to {{n}} more events on {{date}}": "Tautan ke {{n}} acara lainnya pada {{date}}",
  "Link to 1 more event on {{date}}": "Tautan ke 1 acara lainnya pada {{date}}",
  CW: "Minggu {{week}}"
}, Ei = {
  ...xi,
  ...Li
}, Ci = {
  Date: "Datum",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Dal msc",
  "Previous month": "Pedchoz msc",
  "Choose Date": "Vyberte datum"
}, Ti = {
  Today: "Dnes",
  Month: "Msc",
  Week: "Tden",
  Day: "Den",
  "Select View": "Vyberte zobrazen",
  events: "udlosti",
  event: "udlost",
  "No events": "dn udlosti",
  "Next period": "Pt obdob",
  "Previous period": "Pedchoz obdob",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Celodenn a vcedenn udlosti",
  "Link to {{n}} more events on {{date}}": "Odkaz na {{n}} dalch udlost dne {{date}}",
  "Link to 1 more event on {{date}}": "Odkaz na 1 dal udlost dne {{date}}",
  CW: "Tden {{week}}"
}, Oi = {
  ...Ci,
  ...Ti
}, $i = {
  Date: "Kuupev",
  "MM/DD/YYYY": "PP.KK.AAAA",
  "Next month": "Jrgmine kuu",
  "Previous month": "Eelmine kuu",
  "Choose Date": "Vali kuupev"
}, Ai = {
  Today: "Tna",
  Month: "Kuu",
  Week: "Ndal",
  Day: "Pev",
  "Select View": "Vali vaade",
  events: "sndmused",
  event: "sndmus",
  "No events": "Pole sndmusi",
  "Next period": "Jrgmine periood",
  "Previous period": "Eelmine periood",
  to: "kuni",
  "Full day- and multiple day events": "Tispeva- ja mitmepevasndmused",
  "Link to {{n}} more events on {{date}}": "Link {{n}} rohkematele sndmustele kuupeval {{date}}",
  "Link to 1 more event on {{date}}": "Link hele lisasndmusele kuupeval {{date}}",
  CW: "Ndala number {{week}}"
}, Wi = {
  ...$i,
  ...Ai
}, Fi = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, ji = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "       ",
  "Link to {{n}} more events on {{date}}": "  {{n}}    {{date}}",
  "Link to 1 more event on {{date}}": "  1    {{date}}",
  CW: " {{week}}"
}, Ii = {
  ...Fi,
  ...ji
}, Ri = {
  Date: "Datum",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Sledei mesec",
  "Previous month": "Prethodni mesec",
  "Choose Date": "Izaberite datum"
}, Vi = {
  Today: "Danas",
  Month: "Mesec",
  Week: "Nedelja",
  Day: "Dan",
  "Select View": "Odaberite pregled",
  events: "Dogaaji",
  event: "Dogaaj",
  "No events": "Nema dogaaja",
  "Next period": "Naredni period",
  "Previous period": "Prethodni period",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Celodnevni i viednevni dogaaji",
  "Link to {{n}} more events on {{date}}": "Link do jo {{n}} dogaaja na {{date}}",
  "Link to 1 more event on {{date}}": "Link do jednog dogaaja na {{date}}",
  CW: "Nedelja {{week}}"
}, Ui = {
  ...Ri,
  ...Vi
}, Hi = {
  Date: "Data",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Segent mes",
  "Previous month": "Mes anterior",
  "Choose Date": "Selecciona una data"
}, zi = {
  Today: "Avui",
  Month: "Mes",
  Week: "Setmana",
  Day: "Dia",
  "Select View": "Selecciona una vista",
  events: "Esdeveniments",
  event: "Esdeveniment",
  "No events": "Sense esdeveniments",
  "Next period": "Segent perode",
  "Previous period": "Perode anterior",
  to: "a",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Esdeveniments de dia complet i de mltiples dies",
  "Link to {{n}} more events on {{date}}": "Enlla a {{n}} esdeveniments ms el {{date}}",
  "Link to 1 more event on {{date}}": "Enlla a 1 esdeveniment ms el {{date}}",
  CW: "Setmana {{week}}"
}, Ki = {
  ...Hi,
  ...zi
}, Bi = {
  Date: "",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, Ji = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "   ",
  "Link to {{n}} more events on {{date}}": "   {{n}}   {{date}}",
  "Link to 1 more event on {{date}}": "   1  {{date}}",
  CW: " {{week}}"
}, Gi = {
  ...Bi,
  ...Ji
}, Zi = {
  Date: "Data",
  "MM/DD/YYYY": "MMMM-MM-DD",
  "Next month": "Kitas mnuo",
  "Previous month": "Ankstesnis mnuo",
  "Choose Date": "Pasirinkite dat"
}, qi = {
  Today: "iandien",
  Month: "Mnuo",
  Week: "Savait",
  Day: "Diena",
  "Select View": "Pasirinkite vaizd",
  events: "vykiai",
  event: "vykis",
  "No events": "vyki nra",
  "Next period": "Kitas laikotarpis",
  "Previous period": "Ankstesnis laikotarpis",
  to: "iki",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Visos dienos ir keli dien vykiai",
  "Link to {{n}} more events on {{date}}": "Nuoroda  dar {{n}} vykius {{date}}",
  "Link to 1 more event on {{date}}": "Nuoroda  dar 1 vien vyk {{date}}",
  CW: "{{week}} savait"
}, Xi = {
  ...Zi,
  ...qi
}, Qi = {
  Date: "Datum",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Sljedei mjesec",
  "Previous month": "Prethodni mjesec",
  "Choose Date": "Izaberite datum"
}, eo = {
  Today: "Danas",
  Month: "Mjesec",
  Week: "Nedjelja",
  Day: "Dan",
  "Select View": "Odaberite pregled",
  events: "Dogaaji",
  event: "Dogaaj",
  "No events": "Nema dogaaja",
  "Next period": "Sljedei period",
  "Previous period": "Prethodni period",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Cjelodnevni i viednevni dogaaji",
  "Link to {{n}} more events on {{date}}": "Link do jo {{n}} dogaaja na {{date}}",
  "Link to 1 more event on {{date}}": "Link do jo jednog dogaaja na {{date}}",
  CW: "{{week}}. tjedan"
}, to = {
  ...Qi,
  ...eo
}, no = {
  Date: "Datum",
  "MM/DD/YYYY": "MM.DD.YYYY",
  "Next month": "Naslednji mesec",
  "Previous month": "Prejnji mesec",
  "Choose Date": "Izberi datum"
}, ro = {
  Today: "Danes",
  Month: "Mesec",
  Week: "Teden",
  Day: "Dan",
  "Select View": "Izberi pogled",
  events: "dogodki",
  event: "dogodek",
  "No events": "Ni dogodkov",
  "Next period": "Naslednji dogodek",
  "Previous period": "Prejnji dogodek",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Celodnevni in vednevni dogodki",
  "Link to {{n}} more events on {{date}}": "Povezava do {{n}} drugih dogodkov dne {{date}}",
  "Link to 1 more event on {{date}}": "Povezava do e enega dogodka dne {{date}}",
  CW: "Teden {{week}}"
}, io = {
  ...no,
  ...ro
}, oo = {
  Date: "Pivmr",
  "MM/DD/YYYY": "VVVV-KK-PP",
  "Next month": "Seuraava kuukausi",
  "Previous month": "Edellinen kuukausi",
  "Choose Date": "Valitse pivmr"
}, ao = {
  Today: "Tnn",
  Month: "Kuukausi",
  Week: "Viikko",
  Day: "Piv",
  "Select View": "Valitse nkym",
  events: "tapahtumaa",
  event: "tapahtuma",
  "No events": "Ei tapahtumia",
  "Next period": "Seuraava ajanjakso",
  "Previous period": "Edellinen ajanjakso",
  to: "-",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Koko ja usean pivn tapahtumat",
  "Link to {{n}} more events on {{date}}": "Linkki {{n}} listapahtumaan pivmrll {{date}}",
  "Link to 1 more event on {{date}}": "Linkki 1 listapahtumaan pivmrll {{date}}",
  CW: "Viikko {{week}}"
}, so = {
  ...oo,
  ...ao
}, uo = {
  Date: "Data",
  "MM/DD/YYYY": "LL/ZZ/AAAA",
  "Next month": "Luna urmtoare",
  "Previous month": "Luna anterioar",
  "Choose Date": "Alege data"
}, lo = {
  Today: "Astzi",
  Month: "Lun",
  Week: "Sptmn",
  Day: "Zi",
  "Select View": "Selecteaz vizualizarea",
  events: "evenimente",
  event: "eveniment",
  "No events": "Fr evenimente",
  "Next period": "Perioada urmtoare",
  "Previous period": "Perioada anterioar",
  to: "pn la",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Evenimente pe durata ntregii zile i pe durata mai multor zile",
  "Link to {{n}} more events on {{date}}": "Link ctre {{n}} evenimente suplimentare pe {{date}}",
  "Link to 1 more event on {{date}}": "Link ctre 1 eveniment suplimentar pe {{date}}",
  CW: "Sptmna {{week}}"
}, co = {
  ...uo,
  ...lo
};
class _o extends Error {
  constructor(t) {
    super(`Invalid locale: ${t}`);
  }
}
const fo = (e, t) => (n, r) => {
  if (!/^[a-z]{2}-[A-Z]{2}$/.test(e.value) && e.value !== "sr-Latn-RS")
    throw new _o(e.value);
  const i = e.value.replaceAll("-", ""), o = t.value[i];
  if (!o)
    return n;
  let a = o[n] || n;
  return Object.keys(r || {}).forEach((s) => {
    const d = String(r == null ? void 0 : r[s]);
    d && (a = a.replace(`{{${s}}}`, d));
  }), a;
}, ho = {
  Date: "",
  "MM/DD/YYYY": "MM/DD/YYYY",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, po = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "    ",
  "Link to {{n}} more events on {{date}}": "  {{n}}  -{{date}}",
  "Link to 1 more event on {{date}}": "   -{{date}}",
  CW: "{{week}} "
}, vo = {
  ...ho,
  ...po
}, mo = {
  deDE: Sr,
  enUS: Nr,
  itIT: Er,
  enGB: Or,
  svSE: Wr,
  zhCN: Ir,
  zhTW: Ur,
  jaJP: Kr,
  ruRU: Gr,
  koKR: Xr,
  frFR: ti,
  daDK: ii,
  mkMK: bi,
  plPL: si,
  heIL: vo,
  esES: li,
  nlNL: fi,
  ptBR: vi,
  skSK: ki,
  trTR: Si,
  kyKG: Ni,
  idID: Ei,
  csCZ: Oi,
  etEE: Wi,
  ukUA: Ii,
  caES: Ki,
  srLatnRS: Ui,
  srRS: Gi,
  ltLT: Xi,
  hrHR: to,
  slSI: io,
  fiFI: so,
  roRO: co
}, Do = (e = {}) => {
  const t = new br().withFirstDayOfWeek(e.firstDayOfWeek).withLocale(e.locale).withMin(e.min).withMax(e.max).withPlacement(e.placement).withListeners(e.listeners).withStyle(e.style).withTeleportTo(e.teleportTo).withLabel(e.label).withName(e.name).withDisabled(e.disabled).build(), n = new hr().withConfig(t).build();
  return new yr().withConfig(t).withDatePickerState(Dr(t, e.selectedDate)).withTimeUnitsImpl(n).withTranslate(fo(t.locale, M(mo))).build();
}, ko = (e) => {
  const t = Do(e);
  return new cr(t);
}, bo = _t({
  name: "SxDatePicker",
  props: {
    modelValue: {
      type: String,
      required: !0
    },
    config: {
      type: Object,
      default: () => ({
        listeners: {}
      })
    }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t, expose: n }) {
    const r = "sx" + Math.random().toString(36).substr(2, 9);
    let i;
    return n({
      setTheme: (a) => {
        i == null || i.setTheme(a);
      }
    }), Xt(() => e.modelValue, (a) => {
      (i == null ? void 0 : i.value) !== a && i && (i.value = a);
    }), Qt(() => {
      i = ko({
        ...e.config,
        listeners: {
          ...e.config.listeners,
          onChange: (a) => {
            t("update:modelValue", a);
          }
        },
        selectedDate: e.modelValue
      }), i.render(document.getElementById(r));
    }), () => Q("div", {
      id: r,
      class: "sx-vue-date-picker-wrapper"
    });
  }
});
export {
  go as ScheduleXCalendar,
  bo as SxDatePicker
};
